### PR TITLE
Lead Score Form [1/2]

### PR DESCRIPTION
**Pull request by KrekoG**
### Added
- CustomerIntelligenceGateway with LeadScore support (#160168488)

[Delivers [#160168488](https://www.pivotaltracker.com/story/show/160168488)]

### CODE REVIEW

File: src/Gateways/CustomerIntelligenceGateway.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+class CustomerIntelligenceGateway extends AbstractGateway
+{
+    /**
+     * @author GK
+     * @param string $installation
+     * @param string $token
+     * @return array
+     * @throws \WNowicki\Generic\Exception
+     */
+    public function performLeadScore($installation, $body, $token)
```
Comment (wnowicki) [inline src/Gateways/CustomerIntelligenceGateway.php]: - where is docblock for `$body`?

Comment (KrekoG) [reply src/Gateways/CustomerIntelligenceGateway.php]: - added

Comment (wnowicki) [reply src/Gateways/CustomerIntelligenceGateway.php]:
```php
public function performLeadScore($installation, array $body, $token)
```
??

**Comment by KrekoG** [comment]
@wnowicki ready for another look

File: tests/Gateways/CustomerIntelligenceGatewayTest.php [hunk 2]
```diff
@@ -0,0 +1,75 @@
+<?php
+/*
+ * This file is part of the PayBreak/basket package.
+ *
+ * (c) PayBreak <dev@paybreak.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
```
Comment (EvanBarbour3) [inline tests/Gateways/CustomerIntelligenceGatewayTest.php]: Please remove, we dont use this anymore

**Comment by KrekoG** [comment]
@EvanBarbour3 Ready for another look

File: src/Gateways/CustomerIntelligenceGateway.php [hunk 3]
```diff
@@ -0,0 +1,29 @@
+<?php
+
```
Comment (wnowicki) [inline src/Gateways/CustomerIntelligenceGateway.php]: - I think open source header should be here as this is a part of open source package

**Comment by KrekoG** [comment]
@wnowicki added the missing header and made the changes for the faulty ones

**Pull request merged**